### ISSUE TITLE
Parameter --configuration-source does not copy app settings and connection strings

### ISSUE THREAD

**Issue by cornem** [Azure/azure-cli#6638]
I expected the `--configuration-source` parameter on `az webapp deployment slot create` command to copy app settings and connection strings to the new slot, but it doesn't.

Is there another way to achieve this using the CLI? Using the Azure Portal, it is possible to clone the configuration from another slot. It would be really useful if this functionality were also available from the CLI.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: ce9f5bf9-d178-de1a-645b-be6947a7c049
* Version Independent ID: cb7fb876-160c-e6d9-d473-7ac39b930b48
* Content: [az webapp deployment slot](https://docs.microsoft.com/en-us/cli/azure/webapp/deployment/slot?view=azure-cli-latest)
* Content Source: [latest/docs-ref-autogen/webapp/deployment/slot.yml](https://github.com/Azure/azure-docs-cli-python/blob/live/latest/docs-ref-autogen/webapp/deployment/slot.yml)
* Service: **app-service-web**
* GitHub Login: @rloutlaw
* Microsoft Alias: **routlaw**

**Comment by yugangw-msft** [comment]
`--configuration-source` is the way to copy settings over, if it doesn't work then there is a bug. Can you please confirm it is indeed not working. We have a [test](https://github.com/Azure/azure-cli/blob/dev/src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/tests/latest/test_webapp_commands.py#L512-L514) to cover this scenario

**Comment by cornem** [comment]
I have narrowed it down a bit further. The difference between the portal and `--configuration-source` is that the portal also clones app settings or configuration strings that are configured as 'slot settings'. The CLI tools apparently ignore slot settings. 

It would probably be helpful if `--configuration-source` worked exactly like the clone functionality in the portal.

**Comment by panchagnula** [comment]
Let me look at this is details & update the issue.

**Comment by panchagnula** [comment]
@cornem you are referring to having settings marked as SlotSettings correct? 
![image](https://user-images.githubusercontent.com/13839372/43231588-e0878602-9021-11e8-8855-af14fa1a75e9.png)
then yes, we should copy over the settings that are marked as Slot Settings if the app that has these values set is used during slot create.

**Comment by cornem** [comment]
Yes, I'm referring to settings marked as slot setting.

**Comment by panchagnula** [comment]
Will test this & update the thread.

**Comment by panchagnula** [comment]
Talking to our API folks setting the --configuration-source doesn't automatically copy the setting when creating the slot, the client needs to do this explicitly.

**Comment by btardif** [comment]
This might require a bit more work than we can accommodate in the current sprint. moving this to next.

**Comment by rulrok** [comment]
I was expecting the behaviour as described by OP because that is how it works from the azure portal:

![image](https://user-images.githubusercontent.com/3097987/64400876-82a95600-d044-11e9-8934-e246e4452617.png)

Following the azure portal wizard, it copies all appsettings and connection strings (as expected by me).

**Comment by panchagnula** [comment]
@rulrok you are not incorrect in expecting this behavior - it turns out our backend APIs that CLI uses don't automatically copy the settings over - CLI needs to explicitly handle this. We need make the change.

**Comment by yungezz** [comment]
hi @arrownj  could you pls help to follow up on this item also? thanks.

**Comment by yonzhan** [comment]
move to S165.

**Comment by panchagnula** [comment]
Moving to S169

**Comment by awentzel** [comment]
I could really use this right now.  What is S169?  It would be helpful to the community to refer to the non-internal language.  As this is slated to be released in August 2020 or something.  I now have 20 apps to make this change on manually.  ug.

**Comment by awentzel** [comment]
I could really have used this today and looking forward to the release, hoping it does not get pushed into the future much more.

**Comment by johanandrel** [comment]
Any progress on this? Took a while to realize that --configuration-source which is documented as "Source slot to clone configurations from" does not actually clone the configuration.

**Comment by yashvesikar** [comment]
Is there any update on the status of this ticket? The documentation is not clear and it would be a very helpful feature.

**Comment by marsch** [comment]
any updates here?

**Comment by patroza** [comment]
Any news?!

**Comment by panchagnula** [comment]
@JasonFreeberg - need some clarification on this issue - should settings marked slot settings be copied during slot create using configuration source or clone operations? From one of the comments above, portal was copying over all settings during slot create - irrespective of it being slot settings or not. I believe portal client does an explicit copy of all the settings, is this the expected behavior since CLI depends on the API to pick the correct settings. Please comment. Thanks!

**Comment by HerrNiklasRaab** [comment]
I think the cli should replicate the behavior from the portal:

Is a setting marked as deployment slot setting, then it should:
- Not be copied over during a swap
- Should be cloned over when creating a new slot and you specify either in the Portal using the drop-down or with cli with --configuration-source parameter

This is the behavior I expected and was surprised to see that this was not the case.

**Comment by JasonFreeberg** [comment]
Just ran through this on the Portal and compared. I agree with @HerrNiklasRaab, let's use the same implementation as the Portal (copy _all_ settings).

**Comment by Stateofminf** [comment]
Coco

**Comment by sgiddings-kunzleigh** [comment]
I encountered this issue this week for the first time and took me a long time to figure out the problem. I expected the behavior of the Azure CLI to mimic the Azure portal and took me a long time to realize it wasn't. I am trying to build something in a Jenkins pipeline and not having it mimic the Portal functionality is causing problems.

Any update on when this will be implemented and work like the Portal?

**Comment by HerrNiklasRaab** [comment]
This is really a blocker, what is the current plan? I just see this being moved to the next sprint over and over?

**Comment by sgiddings-kunzleigh** [comment]
@HerrNiklasRaab Not sure if this is a solution for you, but what we did in our pipelines is use the Az module for PowerShell. The PowerShell command `New-AzWebAppSlot` with the `-SourceWebApp` parameter mimics the expected behavior of the Azure CLI command being discussed here and the behavior of the Azure Portal.

So of the three management options, it appears the Azure CLI is the outlier.

**Comment by OPthyago** [comment]
Is there any news on this subject?

**Comment by RodrigoSedano** [comment]
I have the same problem using the command below. When I access to the Configuration -> General setting of the slot, the stack and .net version are not set.

`az webapp config set  -g <resource-group> -n <webapp-name> --net-framework-version v5.0 --slot staging`

Any suggesttion?.

**Comment by AgSync-Aaron** [comment]
We just ran into this issue when using the az cli in a ci/cd pipeline and were perplexed on what caused our app to fail, then we discovered this bug. This really should really be tagged as a bug, not a feature request. It appears that the [azure powershell module](https://docs.microsoft.com/en-us/powershell/module/az.websites/new-azwebappslot?view=azps-8.2.0) does copy the configuration.

Please make this a priority.

**Comment by panchagnula** [comment]
> We just ran into this issue when using the az cli in a ci/cd pipeline and were perplexed on what caused our app to fail, then we discovered this bug. This really should be tagged as a bug, not a feature request. It appears that the [azure powershell module](https://docs.microsoft.com/en-us/powershell/module/az.websites/new-azwebappslot?view=azps-8.2.0) does copy the configuration.
> 
> Please make this a priority.

Thanks for the comment - this is a priority on the API end to offer a consistent approach, so that there is no need for each client to maintain custom code & offer different experiences. Honestly, there are certain configurations on app service that shouldn't be copied - the plan to to make the API the single source of truth, so clients don't land into issues copying secure info that is not supposed to shared with slots. Sorry for the inconvenience here & CLI will update this once the API is fixed Thanks!

**Comment by AgSync-Aaron** [comment]
Thank you @panchagnula for the fast reply and clarification! I appreciate the work you all do.

**Comment by alldentobias** [comment]
Also ran into this bug today when setting up auto-created slots through CI/CD. This should really be fixed.

**Comment by cephalin** [comment]
#reassign:btardif

**Comment by cephalin** [comment]
#reassign:btardif

**Comment by btardif** [comment]
@tulikac the slot config clone is done on the client side (in this case the CLI) 

Lets take a look at the portal slot copy behavior and see if we can replicate in CLI.

We should also touch base with control plane to see if this is something they can plug on their end.

**Comment by tuukkapuranen** [comment]
In addition to the issue of not copying slot settings, the use of `--configuration-source` parameter in `az webapp deployment slot create` for an _existing slot_ results in the slot settings of the slot in question to be deleted. Should this behavior be discussed as a part of this issue or reported separately?

**Comment by sijucm** [comment]
Years of people wasting time and Microsoft does not fix anything. Please remove that parameter instead of wasting my time. I

**Comment by t-NuoWenLei** [comment]
Would love some progress for this.

**Comment by walidhai** [comment]
@seligj95 any updates on this?

**Comment by seligj95** [comment]
> [!NOTE]
> **[AI-Assisted Triage]** This comment was generated with AI assistance as part of an App Service CLI issue review. Reply to this comment and tag @seligj95 to follow up.

**Summary**: Issue requires a breaking change to resolve properly.

**Status**: On Hold

**Priority**: P2

**Next Steps**: On hold — this fix requires a breaking change, which can only ship during the bi-annual breaking change window. Will be scheduled for the next available window.
